projects/ui-components/lf-selection-list/resize-column.directive.spec.ts

```diff
@@ -0,0 +1,184 @@
+import { Component } from '@angular/core';
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { ResizeColumnDirective } from './resize-column.directive';
+
+@Component({
+  template: `
+  <div style="position: relative">
+    <table>
+      <tr [style.gridTemplateColumns]="gridTemplateColumns">
+        <th>
+          <div style="width:100%; height: 100%; position: relative" [lfResizeColumn]="true" (widthChanged)="onWidthChange($event)">
+        Header 1</div>
+        </th>
+        <th>Header 2</th>
+        <th>Header 3</th>
+      </tr>
+      <tr [style.gridTemplateColumns]="gridTemplateColumns">
+        <td>1 1</td>
+        <td>1 2</td>
+        <td>1 3</td>
+      </tr>
+      <tr [style.gridTemplateColumns]="gridTemplateColumns">
+        <td>2 1</td>
+        <td>2 2</td>
+        <td>2 3</td>
+      </tr>
+    </table>
+  </div>`,
+  styles: [
+    `
+    table {
+      border-spacing: 0;
+    }
+
+    tr {
+      display: grid;
+    }
+
+    th {
+      padding: 0 10px;
+    }
+
+    .resize-handle {
+    display: inline-block;
+    position: absolute;
+    top: 0;
+    height: 100%;
+    cursor: col-resize;
+    opacity: 0;
+    width: 15px;
+    right: -17.5px;
+  }
+
+  .repository-browser-resize-overlay {
+    position: absolute;
+    top: 0;
+    height: 100%;
+    width: 1px;
+    background-color: #aeaeae;
+    user-select: none;
+    z-index: 1;
+    cursor: col-resize;
+  }
+  `]
+})
+class TestComponent {
+  widthChanged: number = 100;
+  gridTemplateColumns: string = `${this.widthChanged}px 100px 100px`;
+  onWidthChange(width: number) {
+    console.log('width changed', width);
+    this.widthChanged = width;
+    this.gridTemplateColumns = `${this.widthChanged}px 100px 100px`;
+  }
+}
+
+describe('ResizeColumnDirective', () => {
+  let fixture: ComponentFixture<TestComponent>;
+  beforeEach(async () => {
```

Comment:
test if fResizeColumn is false

Reply:
hard to test ngOnInit is only called once in the directive. This should not be a problem because the directive is only for internal use. Will skip this test for now